dev-libs/libflatarray/libflatarray-0.3.0.ebuild

```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit cmake-utils cuda
+
+SRC_URI="http://www.libgeodecomp.org/archive/${P}.tar.bz2"
+KEYWORDS="~amd64 ~ppc ~x86"
+
+DESCRIPTION="Struct of arrays library with object oriented interface for C++"
+HOMEPAGE="http://www.libgeodecomp.org/libflatarray.html"
+
+SLOT="0"
+LICENSE="Boost-1.0"
+IUSE="cuda doc"
+
+DEPEND="
+	doc? ( app-doc/doxygen )
+	cuda? ( dev-util/nvidia-cuda-toolkit )"
```

Comment:
Yes, this is valid.

In the case of LibFlatArray enabling the CUDA build option obviously requires the CUDA toolkit to be installed. If CUDA gets removed after the installation of LibFlatArray, the library stays in a usable state. Its CUDA components only get activated if someone's using NVCC (from the CUDA Toolkit) to compile source.